### ISSUE TITLE
Is that possible to use batchnorm for data parallelisation?

### ISSUE THREAD

**Issue by horserma** [apache/incubator-mxnet#4305]
Is it possible to make batch norm working on the whole batch when doing data split? To be more specific, batch norm calculates the mean and var for the current batch. If for example, we use batch-size=8 and use 4 gpus, then each gpu will have 2 images and the batch norm will only calculates the mean and var based on these 2. Any possibility the batch norm can work on the whole 8 images? @piiswrong @tqchen

**Comment by piiswrong** [comment]
its possible, but complicated. you need you write a cross GPU batchnorm layer

**Comment by horserma** [comment]
In this project [https://github.com/hszhao/PSPNet](url), they implemented this in caffe, but I am not quiet familiar with caffe. It seems pretty promising to let batch norm be real batch norm when doing data parallelisation. Is it difficult to implement it in Mxnet? @piiswrong

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!

**Comment by zhanghang1989** [comment]
MXNet Gluon Cross-GPU Batch Norm is Implemented here 
 https://github.com/zhanghang1989/MXNet-Gluon-SyncBN. Feel free to try and leave comments.
